src/components/ChartSetting.jsx

```diff
@@ -47,156 +47,165 @@ const ThemeToggle = ({
     </div>
 );
 
-const ChartSetting = ({
-    assetInformation,
-    ChartSettingMenu,
-    closeMenu,
-    countdown,
-    historical,
-    isAutoScale,
-    isHighestLowestMarkerEnabled,
-    isMobile,
-    languages,
-    menuOpen,
-    position,
-    selectedLanguage,
-    setAssetInformation,
-    setAutoScale,
-    setHistorical,
-    setLanguage,
-    setPosition,
-    setTheme,
-    setView,
-    showCountdown,
-    view,
-    theme,
-    toggleHighestLowestMarker,
-}) => {
-    const renderMain = () => (
-        <div>
-            <div className="title">
-                <div className="title-text"> {t.translate('Settings')} </div>
-                <CloseIcon className="icon-close-menu" onClick={() => closeMenu()} />
-            </div>
-            <div className="body">
-                <div className="ciq-list ciq-list-setting">
-                    {!isMobile ? <ThemeToggle setPosition={setPosition} position={position} /> : ''}
-                    <div className="ciq-list-item">
-                        <span className="ciq-icon-text">{t.translate('Auto Scale')}</span>
-                        <div className="ciq-action">
-                            <Switch
-                                value={isAutoScale}
-                                onChange={setAutoScale}
-                            />
+class ChartSetting extends Component {
+    componentDidMount() {
+        const { init, activeLanguages } = this.props;
```

Comment:
@shayan-binary What happens by changing `activeLanguages` after mounting the component. Assume the page is loaded completely then the value of `activeLanguages` should be changed.

Reply:
@morteza-binary so you mean. `activeLanguages` may be changed during the rendering?

Reply:
of course, we can make it in a way to update per `activeLanguages` changed, I just thought it's going to initial in the beginning

Reply:
@shayan-binary it'd be better to support it too.